Bugfixed, published in this page:

GlobalManager opened the pull request:
https://github.com/OCA/server-tools/issues/1179

And Chilean translation added.

rockcesar committed:
Bugfixed, published in this page:

https://github.com/OCA/server-tools/issues/1179

And Chilean translation added.

pedrobaeza commented:
You are mixing several things in the same patch. Please change only the needed for solving the issue, and you can change translations in Transifex: https://odoo-community.org/page/translate

rockcesar committed:
Translation rolledback.

rockcesar committed:
PEP8 updated.

rockcesar committed:
PEP8 fixed.

rockcesar committed:
PEP-8 updated.

rockcesar committed:
PEP-8 updated.

rockcesar committed:
PEP-8 updated.

rockcesar committed:
PEP-8 updated.

GlobalManager commented:
Some errors were encountered, related to the version of npm in the server, for example:

/home/travis/.npm/_logs/2018-05-28T01_33_46_147Z-debug.log

GlobalManager commented:
@gurneyalex 
@pedrobaeza 

Watch this. Some errors were encountered, related to the version of npm in the server, for example:

/home/travis/.npm/_logs/2018-05-28T01_33_46_147Z-debug.log

pedrobaeza commented:
The error seems to be due to too many requests.

Any way, I see that you are changing too much things and removing features of the module, so please keep the minimal diff for fixing the problem. If there are several things to fix, only one per PR.

rockcesar committed:
Changes more located.

rockcesar committed:
New changes, oriented to fix auditlog.

rockcesar committed:
http_request updated.

GlobalManager commented:
All fixes are uploaded.

GlobalManager commented:
You can merge whenever you can. Thanks.

GlobalManager commented:
Can you check this?. @pedrobaeza

Thanks

sebalix commented on auditlog/models/http_request.py:
```diff
@@ -47,20 +45,14 @@ def current_http_request(self):
         first call.
         If no HTTP request is available, returns `False`.
         """
+
```
Is this empty line required? It introduces a useless change which can create trouble when backporting patches between branches.

sebalix commented on auditlog/models/http_request.py:
```diff
@@ -70,4 +62,5 @@ def current_http_request(self):
             }
             httprequest.auditlog_http_request_id = self.create(vals).id
             return httprequest.auditlog_http_request_id
+
```
Same remark, is this empty line required?

sebalix commented on auditlog/models/http_request.py:
```diff
@@ -47,20 +45,14 @@ def current_http_request(self):
-                        httprequest.auditlog_http_request_id))
-                if self.env.cr.fetchone():
-                    return httprequest.auditlog_http_request_id
+                return httprequest.auditlog_http_request_id
```
This piece of code was added maybe in 8.0 or 9.0 to avoid some error, are you sure it is not required anymore? You got some problem with it?

sebalix commented on auditlog/models/http_session.py:
```diff
@@ -41,6 +41,7 @@ def current_http_session(self):
         first call.
         If no HTTP user session is available, returns `False`.
         """
+
```
Is this empty line required?

sebalix commented on auditlog/models/http_session.py:
```diff
@@ -57,4 +58,5 @@ def current_http_session(self):
             }
             httpsession.auditlog_http_session_id = self.create(vals).id
             return httpsession.auditlog_http_session_id
+
```
Is this empty line required?

sebalix commented on auditlog/models/rule.py:
```diff
@@ -4,6 +4,9 @@
 
+import logging
+_logger = logging.getLogger(__name__)
+
```
`_logger` does not seem to be used in this file.

sebalix commented on auditlog/models/rule.py:
```diff
@@ -237,28 +239,31 @@ def _make_read(self):
-            if args and isinstance(args[0], sql_db.Cursor):
-                cr, uid, ids = args[0], args[1], args[2]
+            if self and isinstance(self.env.cr, sql_db.Cursor):
+                cr, uid, ids = self.env.cr, self.env.uid, self.ids
```
This code is to be compatible with old API which can always be present in a Odoo 10.0 project (modules not standard nor in OCA), that's why the signature of the method was `*args, **kwargs` to inspect the content of `args`.
If your call to `read(...)` is made with the new API, the rule creation is already satisfied below line 264 ("New API", https://github.com/OCA/server-tools/pull/1259/files#diff-e2cdd2f4b9da04f93936a15fa53fef79R269).

GlobalManager replied:
I can let the signature as *args, but the signature *args, **kwargs, was what was generating the error, fixed by this patch. The actual signature for read in Odoo 10.0, is:

def read(self, fields=None, load='_classic_read'): #Two parameters

I can let the signature like:
def read(self, *args):

So the next sentence doesn't make sense in this case: cr, uid, ids = args[0], args[1], args[2]

I guess read in Odoo 10.0, isn't compatible with old api anymore. The module is working.

sebalix replied:
Okay, I'm testing a piece of code with an `read` overload in old API, and it seems it's not compatible anymore with the new API in Odoo 10.0.
So OK to drop the support for old API. But regarding the fix, the one proposed here is wrong. The `_make_read` method should be written like this:
```python

        @api.multi
        def read(self, fields=None, load='_classic_read'):
            result = read.origin(self, fields, load)
            read_values = dict((d['id'], d) for d in result)
            # If the call came from auditlog itself, skip logging:
            # avoid logs on `read` produced by auditlog during internal
            # processing: read data of relevant records, 'ir.model',
            # 'ir.model.fields'... (no interest in logging such operations)
            if self.env.context.get('auditlog_disabled'):
                return result
            self = self.with_context(auditlog_disabled=True)
            rule_model = self.env['auditlog.rule']
            rule_model.sudo().create_logs(
                self.env.uid, self._name, self.ids,
                'read', read_values, None, {'log_type': log_type})
            return result

        return read
```

sebalix commented on auditlog/models/rule.py:
```diff
@@ -362,30 +367,36 @@ def create_logs(self, uid, res_model, res_ids, method,
+                        old_values)
+                elif method is 'write':
+                    self._create_log_line_on_write(
+                        log, diff.changed(), old_values, new_values)
```
If the only change is the condition `if res_name and res_name != "False"`, limit the diff by rewriting it to use the `continue` statement as we are in a `for` loop.
But this change also need some clarification, have you an example of how you encountered a "False" string value?

GlobalManager replied:
There is, always, a case when creates a register of the action performed, but creates two of them each time, one with (example): res.partned(1,), and the other the same res.partned(1,), but the first the attributes in False, like the name, although it's the object with the correct ID: res.partned(1,).

sebalix replied:
Can you write
```python
if not res_name or res_name == "False":
   continue
```
to avoid an indentation level? It makes the diff very big (and harder to backport too).

sebalix reviewed (changes requested):
For sure some bugs need to be fixed in `auditlog`. Some remarks, but in general we need some explanation to understand the different fixes proposed here.

rockcesar committed:
rule updated.

rockcesar committed:
rule fixed.

rockcesar committed:
rule fixed.

rockcesar committed:
http_request updated.

rockcesar committed:
http_session fixed.

rockcesar committed:
rule updated.

rockcesar committed:
Security fixes. The reads, writes, creates and unlinks were going registered,
even if it wasn't checked. In auditlog 11 is still happening the same, but
I fixed it also and I'll upload it.

rockcesar committed:
rule updated.

rockcesar committed:
rule updated.

rockcesar committed:
rule updated.

GlobalManager commented:
@sebalix We did some changes and fix the feature uploaded.

The reads, writes, creates and unlinks were going to be registered even if the read, create, write, unlink, weren't checked. We fixed this issue.

GlobalManager commented:
To reproduce the issue with the last code of auditlog: Set the rule for res.users and subscribe it, in the rules view, without marking any checkbox. Then perform one read of res.users, that can be made reloading the Odoo's webpage. The read will be registered, even if the read checkbox wasn't checked.

StefanRijnhart commented:
@GlobalManager can you create a separate PR for that last bugfix? BTW. the idea is that if the rule is not enabled for an operation, the model is not patched for that method: https://github.com/OCA/server-tools/blob/10.0/auditlog/models/rule.py#L131 Can you fix the problem there instead?

GlobalManager commented:
But there are 4 checkbox to be checked, if the checkbox aren't checked, the rules shouldn't be executed. Is the normal flow of a understandable module.

rockcesar committed:
rule updated.

GlobalManager commented:
Create patching method is working in the current patch to be merged: https://github.com/OCA/server-tools/blob/10.0/auditlog/models/rule.py#L131

rockcesar committed:
rule updated.

StefanRijnhart commented:
Thanks for isolating that one! Let's continue the discussion on the 11.0 port that you already created here: https://github.com/OCA/server-tools/pull/1264

GlobalManager commented:
Once it is merged, I will apply the others changes, you told me to delete.

GlobalManager commented:
Because, the rest of the changes depend of the changes in this PR.

sebalix commented on auditlog/models/rule.py:
```diff
@@ -237,28 +237,31 @@ def _make_read(self):
                 rule_model = env['auditlog.rule']
-                rule_model.sudo().create_logs(
+                rule_model.sudo().with_context({'auditlog_disabled': True}).\
+                    create_logs(
```
Why changing this? the environment has already the `auditlog_disabled` set to True (two lines above).

sebalix commented on auditlog/models/rule.py:
```diff
@@ -362,27 +365,34 @@ def create_logs(self, uid, res_model, res_ids, method,
+                'http_request_id': http_request_model.
+                current_http_request(),
+                'http_session_id': http_session_model.
+                current_http_session(),
```
Do not change these lines, this is not necessary.

sebalix commented on auditlog/models/rule.py:
```diff
@@ -362,27 +365,34 @@ def create_logs(self, uid, res_model, res_ids, method,
             if method is 'create':
-                self._create_log_line_on_create(log, diff.added(), new_values)
+                self._create_log_line_on_create(
+                    log, diff.added(), new_values)
```
Do not change these lines, this is not necessary.

sebalix commented on auditlog/models/rule.py:
```diff
@@ -362,27 +365,34 @@ def create_logs(self, uid, res_model, res_ids, method,
                 self._create_log_line_on_read(
-                    log, old_values.get(res_id, EMPTY_DICT).keys(), old_values)
+                    log, old_values.get(res_id, EMPTY_DICT).keys(),
+                    old_values)
```
Do not change these lines, this is not necessary.

sebalix reviewed (changes requested):
Comment done bellow.
Please avoid to change lines of code not related to the fix, it makes the diff bigger (hard to read, and possibly hard to backport in other branches).

rockcesar committed:
rule modified.

rockcesar committed:
rule updated.

rockcesar committed:
rule updated.

rockcesar committed:
auditlog_view.xml updated.

rockcesar committed:
auditlog_view updated.

GlobalManager commented:
The test_pylint applied during the tests, is giving the error, but there is not have to be with auditlog.

rockcesar committed:
rule updated.

GlobalManager commented:
What we have to

GlobalManager commented:
If the pylint failed, what can we do?.

sebalix commented:
The lint failed but it's not related to auditlog, so do nothing about it. If a lint was related to auditlog, just fix it :)

GlobalManager commented:
That's nice.

sebalix commented on auditlog/models/rule.py:
```diff
@@ -237,44 +237,26 @@ def _make_read(self):
             result2 = result
             if not isinstance(result2, list):
                 result2 = [result]
             read_values = dict((d['id'], d) for d in result2)
```
`result`/`result2` trick isn't necessary anymore as the `read` method is only compatible with the new API. Check the piece of code I sent in the other comment.
